File: app/src/main/java/com/github/colinjeremie/willyoufindit/adapters/TutoAdapter.java
package com.github.colinjeremie.willyoufindit.adapters;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import com.github.colinjeremie.willyoufindit.R;
import com.github.colinjeremie.willyoufindit.fragments.BaseTutoFragment;
import com.github.colinjeremie.willyoufindit.fragments.PresentationFragmentStep3;

/**
 * * WillYouFindIt
 * Created by jerem_000 on 4/23/2016.
 */
public class TutoAdapter extends FragmentStatePagerAdapter {
    public static final int COUNT = 3;

    public TutoAdapter(FragmentManager pFragmentManager) {
        super(pFragmentManager);
    }

    @Override
    public Fragment getItem(int position) {
        if (position == 0){
            return BaseTutoFragment.getInstance(R.layout.fragment_presentation_step1);
        } else if (position == 1){
            return BaseTutoFragment.getInstance(R.layout.fragment_presentation_step2);
        }
        return new PresentationFragmentStep3();
    }

    @Override
    public int getCount() {
        return COUNT;
    }
}

File: app/src/main/java/com/github/colinjeremie/willyoufindit/adapters/TutoAdapter.kt
package com.github.colinjeremie.willyoufindit.adapters

import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter
import com.github.colinjeremie.willyoufindit.R
import com.github.colinjeremie.willyoufindit.fragments.BaseTutoFragment
import com.github.colinjeremie.willyoufindit.fragments.PresentationFragmentStep3

class TutoAdapter(fragmentManager: FragmentManager?) : FragmentStatePagerAdapter(fragmentManager) {

    companion object {
        const val COUNT = 3
        private const val STEP1_POSITION = 0
        private const val STEP2_POSITION = 1
    }

    override fun getItem(position: Int) =
            when (position) {
                STEP1_POSITION -> BaseTutoFragment.getInstance(R.layout.fragment_presentation_step1)
                STEP2_POSITION -> BaseTutoFragment.getInstance(R.layout.fragment_presentation_step2)
                else -> PresentationFragmentStep3()
            }

    override fun getCount() = COUNT
}